packages/datadog-instrumentations/src/mariadb.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+
+const shimmer = require('../../datadog-shimmer')
+
+const startCh = channel('apm:mariadb:query:start');
+const finishCh = channel('apm:mariadb:query:finish');
+const errorCh = channel('apm:mariadb:query:error');
+
+addHook(
+  {
+    name: 'mariadb',
+    file: 'lib/connection-promise.js',
+    versions: ['>=3'],
+  },
+  (ConnectionPromise) => {
+    shimmer.wrap(ConnectionPromise, '_QUERY_CMD', (queryCmd) => {
```

Comment:
It looks like both `ConnectionPromise` and `ConnectionCallback` simply call the underlying `Connection` methods directly with `resolve` and `reject` callbacks. Instrumenting `Connection` instead would make it easier to support both classes with only 1 hook.

Reply:
Here it ultimately calls the underlying `Connection` object: https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/lib/connection-promise.js#L96. `ConnectionCallback` also calls it in the same way but passes `resolve` and `reject` manually.

Reply:
You're right. Good catch.